portal/views.py

```diff
@@ -169,12 +217,62 @@ def admin_panel(request):
 
     form = CreateTestForm()
 
-    return render(request, 'portal/admin.html', {
+    return render(request, 'portal/admin_2.html', {
         "tests": test,
         "form": form,
         "is_form_valid": is_form_invalid,
+        "test_number": no_tests,
+    })
+
+def test_activate(request, testID):
+    if (not request.user.is_authenticated):
+        return HttpResponseRedirect(reverse('login'))
+
+    if (not request.user.is_superuser):
+        return HttpResponseForbidden('You are not allowed to access this resource!')
+    
+    test = Test.objects.get(id=testID)
+
+    if(test.activated):
+        test.activated = False
+    else:
+        test.activated = True
+    
+    test.save()
+
+    return HttpResponseRedirect(reverse('basic-settings', args=[testID]))
+
+def time_settings(request, testID):
+    if (not request.user.is_authenticated):
+        return HttpResponseRedirect(reverse('login'))
+
+    if (not request.user.is_superuser):
+        return HttpResponseForbidden('You are not allowed to access this resource!')
+
+    test = Test.objects.filter(id=testID).first()
+
+    return render(request, 'portal/test-settings/time-settings.html', {
+        'test': test
     })
 
+def signup_user(request):
```

Comment:
why have you not added the required html file in the render here for signup? Please render the requested html file.

Reply:
The signup HTML file has not been added by the frontend team yet, I will add it as soon as the file is available.

Reply:
Ok, then for now we will  merge this branch.